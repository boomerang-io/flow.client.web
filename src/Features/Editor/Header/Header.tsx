import React from "react";
import { Link, useRouteMatch } from "react-router-dom";
import { Breadcrumb, BreadcrumbItem, Button } from "@carbon/react";
import {
  ConfirmModal,
  ComposedModal,
  FeatureHeader as Header,
  FeatureHeaderTitle as HeaderTitle,
  FeatureNavTab as Tab,
  FeatureNavTabs as Tabs,
} from "@boomerang-io/carbon-addons-boomerang-react";
import VersionCommentForm from "./VersionCommentForm";
import VersionSwitcher from "./VersionSwitcher";
import { appLink } from "Config/appConfig";
import { Add, DocumentExport } from "@carbon/react/icons";
import { AxiosResponse } from "axios";
import { UseMutationResult } from "react-query";
import {
  ModalTriggerProps,
  ComposedModalChildProps,
  WorkflowCanvasState,
  WorkflowView,
  WorkflowViewType,
  ChangeLog,
} from "Types";
import styles from "./header.module.scss";

interface DesignerHeaderProps {
  changeLog: ChangeLog;
  createRevision: (reason: string, callback?: () => any) => void;
  changeRevision: (revisionNumber: string) => void;
  isOnDesigner: boolean;
  revisionMutator: UseMutationResult<AxiosResponse<any, any>, unknown, { workflowId: any; body: any }, unknown>;
  revisionState: WorkflowCanvasState;
  viewType: WorkflowViewType;
  revisionCount: number;
}

const DesignerHeader: React.FC<DesignerHeaderProps> = ({
  createRevision,
  changeRevision,
  isOnDesigner,
  revisionCount,
  revisionMutator,
  revisionState,
  viewType,
}) => {
  const routeMatch: { params: { teamId: string; workflowId: string } } = useRouteMatch();
  const {
    params: { teamId, workflowId },
  } = routeMatch;
  const { name } = revisionState;
  const { version: currentRevision } = revisionState;
  const isPreviousVersion = currentRevision < revisionCount;
  const performActionButtonText = currentRevision < revisionCount ? "Set version to latest" : "Create new version";

  // Need to hardcode that the version is being reset in the change log for now based on the current implementation
  const resetVersionToLatestWithMessage = () => {
    createRevision(`Set ${currentRevision} to the latest version`);
  };

  return (
    <Header
      className={styles.container}
      nav={
        <Breadcrumb noTrailingSlash>
          <BreadcrumbItem>
            {viewType === WorkflowView.Workflow ? (
              <Link to={appLink.workflows({ teamId })}>Workflows</Link>
            ) : (
              <Link to={appLink.templateWorkflows()}>Template Workflows</Link>
            )}
          </BreadcrumbItem>
          <BreadcrumbItem isCurrentPage>
            <p>{name}</p>
          </BreadcrumbItem>
        </Breadcrumb>
      }
      header={<HeaderTitle>Editor</HeaderTitle>}
      footer={
        <Tabs ariaLabel="Editor pages">
          <Tab label="Workflow" to={appLink.editorDesigner({ teamId, workflowId })} />
          <Tab label="Parameters" to={appLink.editorProperties({ teamId, workflowId })} />
          <Tab label="Configure" to={appLink.editorConfigure({ teamId, workflowId })} />
          <Tab label="Schedule" to={appLink.editorSchedule({ teamId, workflowId })} />
          <Tab label="Change Log" to={appLink.editorChangelog({ teamId, workflowId })} />
        </Tabs>
      }
      actions={
        <section className={styles.workflowButtons}>
          <VersionSwitcher
            currentRevision={currentRevision}
            disabled={!isOnDesigner}
            onChangeVersion={changeRevision}
            revisionCount={revisionCount}
          />
          <div className={styles.workflowActionContainer}>
            <>
              <ConfirmModal
                affirmativeAction={resetVersionToLatestWithMessage}
                children="A new version will be created"
                title={`Set version ${currentRevision} to be the latest`}
                modalTrigger={({ openModal }: ModalTriggerProps) => (
                  <Button
                    disabled={!isOnDesigner}
                    iconDescription="Set version to latest"
                    kind="ghost"
                    onClick={openModal}
                    renderIcon={DocumentExport}
                    size="md"
                    style={!isPreviousVersion ? { display: "none" } : null}
                  >
                    {performActionButtonText}
                  </Button>
                )}
              />
              <ComposedModal
                composedModalProps={{ containerClassName: styles.versionCommentModalContainer }}
                modalHeaderProps={{
                  title: "Create New Version",
                  subtitle: "Enter a comment for record keeping",
                }}
                modalTrigger={({ openModal }: ModalTriggerProps) => (
                  <Button
                    disabled={!isOnDesigner}
                    iconDescription="Create new version"
                    kind="ghost"
                    onClick={openModal}
                    renderIcon={Add}
                    size="md"
                    style={isPreviousVersion ? { display: "none" } : null}
                  >
                    {performActionButtonText}
                  </Button>
                )}
              >
                {({ closeModal }: ComposedModalChildProps) => (
                  <VersionCommentForm
                    closeModal={closeModal}
                    createRevision={createRevision}
                    revisionMutator={revisionMutator}
                  />
                )}
              </ComposedModal>
            </>
          </div>
        </section>
      }
    />
  );
};

export default DesignerHeader;
