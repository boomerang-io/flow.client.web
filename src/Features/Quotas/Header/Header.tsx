import React from "react";
import {
  Button,
  ComposedModal,
  FeatureHeader,
  FeatureHeaderTitle as HeaderTitle,
  InlineNotification,
  Loading,
  ModalBody,
  ModalFooter,
  ModalForm,
  notify,
  ToastNotification,
} from "@carbon/react";
import { useMutation, useQueryClient } from "react-query";
import { resolver, serviceUrl } from "Config/servicesConfig";
import { Reset } from "@carbon/react/icons";
import { ModalTriggerProps, FlowTeam, ComposedModalChildProps, FlowTeamQuotas } from "Types";
import styles from "./Header.module.scss";

interface HeaderProps {
  defaultQuotasData: FlowTeamQuotas;
  defaultQuotasError: object;
  defaultQuotasIsLoading: boolean;
  selectedTeam: FlowTeam;
}

const Header: React.FC<HeaderProps> = ({
  selectedTeam,
  defaultQuotasData,
  defaultQuotasError,
  defaultQuotasIsLoading,
}) => {
  return (
    <FeatureHeader
      className={styles.featureHeader}
      header={<HeaderTitle>{selectedTeam.name}</HeaderTitle>}
      actions={
        <ComposedModal
          composedModalProps={{
            containerClassName: styles.modalContainer,
          }}
          modalHeaderProps={{
            title: "Restore defaults",
            subtitle: "This will change all quotas to the following default values. This action cannot be undone.",
          }}
          modalTrigger={({ openModal }: ModalTriggerProps) => (
            <Button className={styles.resetButton} size="md" renderIcon={Reset} onClick={openModal}>
              Restore defaults
            </Button>
          )}
        >
          {({ closeModal }: ComposedModalChildProps) => (
            <RestoreModalContent
              closeModal={closeModal}
              defaultQuotas={defaultQuotasData}
              defaultQuotasError={defaultQuotasError}
              defaultQuotasIsLoading={defaultQuotasIsLoading}
              teamId={selectedTeam.id}
            />
          )}
        </ComposedModal>
      }
    />
  );
};

interface restoreDefaultProps {
  closeModal: Function;
  defaultQuotas: FlowTeamQuotas;
  defaultQuotasError: object;
  defaultQuotasIsLoading: boolean;
  teamId: string;
}

const RestoreModalContent: React.FC<restoreDefaultProps> = ({
  closeModal,
  defaultQuotas,
  defaultQuotasError,
  defaultQuotasIsLoading,
  teamId,
}) => {
  const cancelRequestRef = React.useRef<{} | null>();
  const queryClient = useQueryClient();

  const { mutateAsync: defaultQuotasMutator, isLoading, error } = useMutation(
    (args: { id: string }) => {
      const { promise, cancel } = resolver.putTeamQuotasDefault(args);
      if (cancelRequestRef?.current) {
        cancelRequestRef.current = cancel;
      }
      return promise;
    },
    {
      onSuccess: () => {
        queryClient.invalidateQueries(serviceUrl.getTeamQuotas({ id: teamId }));
        queryClient.invalidateQueries(serviceUrl.getTeams());
      },
    }
  );

  const handleRestoreDefaultQuota = async () => {
    try {
      await defaultQuotasMutator({ id: teamId });
      closeModal();
      notify(
        <ToastNotification
          kind="success"
          title="Restore Default Quotas"
          subtitle="Successfully restored default quotas"
        />
      );
    } catch {
      notify(<ToastNotification kind="error" title="Something's wrong" subtitle="Failed to restore default quotas" />);
    }
  };

  let buttonText = "Save";
  if (isLoading) {
    buttonText = "Saving...";
  } else if (error) {
    buttonText = "Try again";
  }
  return (
    <ModalForm>
      <ModalBody className={styles.modalBodyContainer}>
        <hr className={styles.divider} />
        {defaultQuotasIsLoading ? (
          <Loading />
        ) : (
          <div className={styles.gridContainer}>
            <section>
              <dt className={styles.detailedTitle}>Maximum number of Workflows </dt>
              <dt className={styles.detailedData}>
                {defaultQuotasError ? "---" : `${defaultQuotas.maxWorkflowCount} Workflows`}{" "}
              </dt>
            </section>
            <section>
              <dt className={styles.detailedTitle}>Maximum Workflow executions </dt>
              <dt className={styles.detailedData}>
                {defaultQuotasError ? "---" : `${defaultQuotas.maxWorkflowExecutionMonthly} per month`}
              </dt>
            </section>
            <section>
              <dt className={styles.detailedTitle}>Storage limit</dt>
              <dt className={styles.detailedData}>
                {defaultQuotasError ? "---" : `${defaultQuotas.maxWorkflowExecutionMonthly}GB per Workflow`}
              </dt>
            </section>
            <section>
              <dt className={styles.detailedTitle}>Maximum Workflow duration</dt>
              <dt className={styles.detailedData}>{`${defaultQuotas.maxWorkflowExecutionTime} minutes`}</dt>
            </section>
            <section>
              <dt className={styles.detailedTitle}>Maximum concurrent Workflows</dt>
              <dt className={styles.detailedData}>
                {defaultQuotasError ? "---" : `${defaultQuotas.maxConcurrentWorkflows} Workflows`}
              </dt>
            </section>
          </div>
        )}
        {error && (
          <InlineNotification
            lowContrast
            kind="error"
            title="Quota restore default failed!"
            subtitle="Give it another go or try again later."
          />
        )}
      </ModalBody>
      <ModalFooter>
        <Button kind="secondary" type="button" onClick={closeModal}>
          Cancel
        </Button>
        <Button disabled={isLoading} onClick={handleRestoreDefaultQuota}>
          {buttonText}
        </Button>
      </ModalFooter>
    </ModalForm>
  );
};

export default Header;
