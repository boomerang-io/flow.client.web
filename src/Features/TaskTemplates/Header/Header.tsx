import React from "react";
import { useParams } from "react-router-dom";
import { Button, InlineNotification, ModalBody, ModalFooter, Tag, TextArea } from "@carbon/react";
import {
  ComposedModal,
  ConfirmModal,
  FeatureHeader,
  Loading,
  ModalFlowForm,
  FeatureNavTab as Tab,
  FeatureNavTabs as Tabs,
  TooltipHover,
} from "@boomerang-io/carbon-addons-boomerang-react";
import VersionHistory from "Components/VersionHistory";
import VersionSwitcher from "./VersionSwitcher";
import moment from "moment";
import { appLink } from "Config/appConfig";
import { serviceUrl } from "Config/servicesConfig";
import { taskIcons } from "Utils/taskIcons";
import { TemplateRequestType, FormProps } from "../constants";
import { Bee, Download, Save, Undo, Reset, ViewOff } from "@carbon/react/icons";
import { FormikProps } from "formik";
import { ComposedModalChildProps, ModalTriggerProps, TaskTemplate } from "Types";
import styles from "./header.module.scss";

interface SaveModalProps {
  cancelRequestRef: any;
  formikProps: FormikProps<FormProps>;
  handleSubmit: (
    values: any,
    resetForm: () => void,
    requestType: string,
    setRequestError: ({ title, subtitle }: { title: string; subtitle: string }) => void,
    closeModal: () => void
  ) => void;
  isLoading: boolean;
  canEdit: boolean;
}

const SaveModal: React.FC<SaveModalProps> = ({ cancelRequestRef, formikProps, handleSubmit, isLoading, canEdit }) => {
  const [requestError, setRequestError] = React.useState<{ title?: string; subtitle?: string } | null>(null);
  const SaveMessage = () => {
    return (
      <>
        <p className={styles.saveConfirmModalText}>
          Choose to overwrite the current version, or save as a new version.
        </p>
        <p className={styles.saveConfirmModalText}>
          <strong>Overwritting the current version</strong> means that this task will automatically be updated in any
          workflow. Do this with extreme caution, and only when you haven’t made any breaking changes.
        </p>
        <p className={styles.saveConfirmModalText}>
          <strong>Saving as a new version</strong> will not automatically update this task in workflows. Users who have
          this task in a workflow will get a notification about the new version and can choose to update or not.
        </p>
      </>
    );
  };
  return (
    <ComposedModal
      modalHeaderProps={{
        title: "Save changes",
      }}
      composedModalProps={{ containerClassName: styles.saveContainer }}
      onCloseModal={() => {
        if (cancelRequestRef.current) cancelRequestRef.current();
      }}
      modalTrigger={({ openModal }: ModalTriggerProps) => (
        <TooltipHover direction="bottom" tooltipText={"Save a new version or update the current one"}>
          <Button
            className={styles.mainActionButton}
            disabled={!formikProps.isValid || !formikProps.dirty}
            size="md"
            renderIcon={Save}
            iconDescription="Save a new version or update the current one"
            onClick={openModal}
          >
            Save...
          </Button>
        </TooltipHover>
      )}
    >
      {({ closeModal }: ComposedModalChildProps) => {
        return (
          <ModalFlowForm>
            {isLoading && <Loading />}
            <ModalBody>
              <SaveMessage />
              <TextArea
                data-testid="save-comments"
                id="comments"
                name="comments"
                key="newTemplateComments"
                labelText="Comments (required for new versions)"
                onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                  formikProps.setFieldValue("comments", e.target.value)
                }
                placeholder="Release notes for the new version"
                style={{ resize: "none" }}
                value={formikProps.values.comments}
              />
              {Boolean(requestError) && (
                <InlineNotification
                  lowContrast
                  style={{ marginBottom: "0.5rem" }}
                  kind="error"
                  title={requestError?.title}
                  subtitle={requestError?.subtitle}
                  onCloseButtonClick={() => setRequestError(null)}
                />
              )}
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" type="button" onClick={closeModal}>
                Cancel
              </Button>
              <Button
                kind="secondary"
                onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
                  e.preventDefault();
                  handleSubmit(
                    formikProps.values,
                    formikProps.resetForm,
                    TemplateRequestType.Overwrite,
                    setRequestError,
                    closeModal
                  );
                }}
              >
                Overwrite this version
              </Button>
              <Button
                onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
                  e.preventDefault();
                  handleSubmit(
                    formikProps.values,
                    formikProps.resetForm,
                    TemplateRequestType.New,
                    setRequestError,
                    closeModal
                  );
                }}
                disabled={!Boolean(formikProps.values.comments)}
              >
                Save new version
              </Button>
            </ModalFooter>
          </ModalFlowForm>
        );
      }}
    </ComposedModal>
  );
};

interface HeaderProps {
  editVerifiedTasksEnabled: boolean;
  cancelRequestRef: object;
  currentRevision: { version: number };
  formikProps: FormikProps<FormProps>;
  handleSaveTaskTemplate: (
    values: any,
    resetForm: () => void,
    requestType: string,
    setRequestError?: (error: any) => void,
    closeModal?: () => void
  ) => void;
  handleputRestoreTaskTemplate: () => void;
  isActive: boolean;
  isLoading: boolean;
  isOldVersion: boolean;
  selectedTaskTemplate: TaskTemplate;
}

const Header: React.FC<HeaderProps> = ({
  editVerifiedTasksEnabled,
  selectedTaskTemplate,
  currentRevision,
  formikProps,
  handleSaveTaskTemplate,
  handleputRestoreTaskTemplate,
  isOldVersion,
  isActive,
  isLoading,
  cancelRequestRef,
}) => {
  const params: any = useParams();
  const TaskIcon = taskIcons.find((icon) => icon.name === selectedTaskTemplate.icon);
  const revisionCount = selectedTaskTemplate.revisions.length;
  const lastUpdated = selectedTaskTemplate?.revisions[revisionCount - 1]?.changelog ?? {};
  const changelogs = selectedTaskTemplate?.revisions?.map((revision) => ({
    ...revision.changelog,
    date: moment(revision?.changelog?.date)?.format("MMM DD, YYYY") ?? "---",
    version: revision.version,
  }));
  changelogs.reverse();
  const canEdit = !selectedTaskTemplate?.verified || (editVerifiedTasksEnabled && selectedTaskTemplate?.verified);

  return (
    <FeatureHeader
      className={styles.featureHeader}
      footer={
        <Tabs ariaLabel="Task template views">
          <Tab
            exact
            label="Overview"
            to={appLink.taskTemplateDetail({ id: selectedTaskTemplate.id, version: currentRevision.version })}
          />
          <Tab
            exact
            label="YAML"
            to={appLink.taskTemplateEditor({ id: selectedTaskTemplate.id, version: currentRevision.version })}
          />
        </Tabs>
      }
      actions={
        <div className={styles.buttons}>
          <VersionSwitcher
            revisions={selectedTaskTemplate.revisions}
            currentRevision={currentRevision}
            revisionCount={revisionCount}
            canEdit={canEdit}
          />
          {!isOldVersion && isActive && (
            <ConfirmModal
              affirmativeAction={formikProps.resetForm}
              affirmativeText="Reset changes"
              children="You are about to reset to the last save of this version, all unsaved changes will be erased. This action cannot be undone, are you sure you want to reset to the latest save?"
              title="Reset changes"
              modalTrigger={({ openModal }: ModalTriggerProps) => (
                <TooltipHover direction="bottom" tooltipText={"Restore the last save of this version"}>
                  <Button
                    className={styles.resetButton}
                    disabled={!formikProps.dirty || !canEdit}
                    size="md"
                    kind="ghost"
                    renderIcon={Undo}
                    onClick={openModal}
                  >
                    {" "}
                    Reset changes
                  </Button>
                </TooltipHover>
              )}
            />
          )}
          {isOldVersion ? (
            <ConfirmModal
              affirmativeAction={() =>
                handleSaveTaskTemplate(formikProps.values, formikProps.resetForm, TemplateRequestType.Copy)
              }
              children={
                <>
                  <p className={styles.confirmModalText}>Sometimes revisiting the past is a good thing.</p>
                  <p
                    className={styles.confirmModalText}
                  >{`This action will create a new version that’s an exact copy of Version ${
                    currentRevision.version
                  }, but it shall be named Version ${revisionCount + 1}. Make sure this is what you want to do.`}</p>
                </>
              }
              affirmativeText="Copy to new version"
              title="Copy to new version"
              modalTrigger={({ openModal }: ModalTriggerProps) => (
                <TooltipHover direction="bottom" tooltipText={"Copy this version to a new version to enable editing"}>
                  <Button
                    className={styles.copyButton}
                    size="md"
                    kind="ghost"
                    renderIcon={Undo}
                    onClick={openModal}
                    disabled={!canEdit}
                  >
                    Copy to new version
                  </Button>
                </TooltipHover>
              )}
            />
          ) : isActive ? (
            <SaveModal
              cancelRequestRef={cancelRequestRef}
              formikProps={formikProps}
              handleSubmit={handleSaveTaskTemplate}
              isLoading={isLoading}
              canEdit={canEdit}
            />
          ) : (
            <ConfirmModal
              affirmativeAction={handleputRestoreTaskTemplate}
              children={
                <>
                  <p className={styles.confirmModalText}>
                    Restoring a task will remove it from the Archive and make it visible in the Workflow Editor, as the
                    most recent version.
                  </p>
                  <p className={styles.confirmModalText}>Are you sure you’d like to restore this task?</p>
                </>
              }
              affirmativeText="Restore this task"
              title="Restore"
              modalTrigger={({ openModal }: ModalTriggerProps) => (
                <Button className={styles.mainActionButton} size="md" renderIcon={Reset} onClick={openModal}>
                  Restore
                </Button>
              )}
            />
          )}
        </div>
      }
    >
      <p className={styles.category}>{selectedTaskTemplate.category}</p>
      <div className={styles.infoContainer}>
        {TaskIcon ? (
          <TaskIcon.Icon style={{ minWidth: "1.5rem", minHeight: "1.5rem", marginRight: "0.75rem" }} />
        ) : (
          <Bee alt={`${selectedTaskTemplate.name} icon`} className={styles.icon} />
        )}
        <h1 className={styles.taskName} title={selectedTaskTemplate.name}>
          {selectedTaskTemplate.name}
        </h1>
        {!isActive && (
          <Tag className={styles.archivedTag} type="gray">
            <ViewOff style={{ marginRight: "0.5rem" }} />
            Archived
          </Tag>
        )}
        <VersionHistory changelogs={changelogs} />
        <TooltipHover direction="right" content="Export latest saved revision of the YAML">
          <a
            className={styles.exportYaml}
            href={serviceUrl.getTaskTemplateYaml({ id: params.id, revision: params.version })}
            download={`${selectedTaskTemplate.name}.yaml`}
            target="_blank"
            rel="noopener noreferrer"
          >
            <Download />
          </a>
        </TooltipHover>
      </div>
      <p className={styles.lastUpdate}>{`Version ${revisionCount === 1 ? "created" : "updated"} ${moment(
        lastUpdated.date
      ).format("MMM DD, YYYY")} by ${lastUpdated.userName ?? "---"}`}</p>
    </FeatureHeader>
  );
};

export default Header;
