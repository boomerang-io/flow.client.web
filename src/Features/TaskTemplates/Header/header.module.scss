@import "Styles/mixins";

.featureHeader {
  align-items: center;
  display: flex;
  min-height: 9.3125rem;
  width: 100%;
  padding: 1.5rem 1.5rem 1rem;
}

.saveContainer {
  max-height: 33rem;
}

.confirmModalText {
  font-size: 0.875rem;
  margin-bottom: 1rem;
}
.saveConfirmModalText {
  font-size: 0.875rem;
  margin-bottom: 1rem;
  width: 36rem;
}

.archivedTag {
  background-color: var(--cds-icon-02);
  color: var(--cds-ui-02);
  margin-left: 1rem !important;
  svg {
    fill: var(--cds-ui-02);
  }
}
.buttons {
  align-self: flex-end;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-end;
  flex-grow: 1;
  > * {
    margin: 1rem 0 0 1.5rem;
  }
}

.copyButton {
  width: 12.8125rem;
}

.resetButton {
  width: 10.875rem;
}

.mainActionButton {
  width: 7.75rem;
}

.category {
  font-size: 0.875rem;
  line-height: 1.125rem;
  margin-bottom: 0.5rem;
}

.taskName {
  font-size: 1rem;
  line-height: 1.25rem;
  margin: 0rem;
  @include multiline-text-overflow-ellipsis();
}
.infoContainer {
  align-items: center;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  :global .cds--assistive-text {
    left: -2rem !important;
  }
}
.info {
  align-items: center;
  display: flex;
  flex-grow: 1;
  overflow: hidden;
}
.icon {
  height: 1.5rem;
  margin-right: 0.75rem;
}
.content {
  align-items: center;
  display: flex;
  justify-content: flex-start;
  width: 100%;
}
.title {
  font-size: 1.5rem;
  font-weight: 600;
  line-height: 2rem;
  margin: 0;
}
.subtitle {
  font-size: 2rem;
  font-weight: 600;
  line-height: 2.5rem;
}

.exportYaml {
  align-items: center;
  color: var(--cds-text-01);
  cursor: pointer;
  display: flex !important;
  justify-content: center;
  margin-left: 0.5rem;
  &:focus {
    @include focus-outline();
  }
}

.lastUpdate {
  color: var(--cds-text-02);
  font-size: 0.75rem;
  margin-bottom: 0.5rem;
  margin-top: 0.75rem;
}
