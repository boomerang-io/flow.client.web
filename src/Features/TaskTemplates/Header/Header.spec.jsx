import React from "react";
import Header from "./index";
import { tasktemplate } from "ApiServer/fixtures";

const mockfn = jest.fn();

const props = {
  editVerifiedTasksEnabled:true,
  formikProps: {},
  selectedTaskTemplate: tasktemplate[0],
  currentRevision: { version: 1 },
  handleSaveTaskTemplate: mockfn,
  handleputRestoreTaskTemplate: mockfn,
  isOldVersion: false,
  isActive: true,
  isLoading: false,
};

describe("Header --- Snapshot", () => {
  it("Capturing Snapshot of Task Templates", async () => {
    const { baseElement } = rtlContextRouterRender(<Header {...props}/>);
    expect(baseElement).toMatchSnapshot();
  });
});

