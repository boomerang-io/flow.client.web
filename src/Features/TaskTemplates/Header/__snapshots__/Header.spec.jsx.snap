// Vitest Snapshot v1

exports[`Header --- Snapshot > Capturing Snapshot of Task Templates 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="cds--bmrg-feature-header _featureHeader_4b0ee6 --bordered"
    >
      <section
        class="cds--bmrg-feature-header__content"
      >
        <p
          class="_category_4b0ee6"
        >
          communication
        </p>
        <div
          class="_infoContainer_4b0ee6"
        >
          <svg
            alt="Task node type get, read"
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            style="min-width: 1.5rem; min-height: 1.5rem; margin-right: 0.75rem;"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
            />
          </svg>
          <h1
            class="_taskName_4b0ee6"
            title="Slack User Look Up"
          >
            Slack User Look Up
          </h1>
          <button
            class="_button_61c4b8"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M26,2H8A2,2,0,0,0,6,4V8H4v2H6v5H4v2H6v5H4v2H6v4a2,2,0,0,0,2,2H26a2,2,0,0,0,2-2V4A2,2,0,0,0,26,2Zm0,26H8V24h2V22H8V17h2V15H8V10h2V8H8V4H26Z"
              />
              <path
                d="M14 8H22V10H14zM14 15H22V17H14zM14 22H22V24H14z"
              />
            </svg>
          </button>
          <a
            class="_exportYaml_4b0ee6"
            download="Slack User Look Up.yaml"
            href="/api/workflow/tasktemplate/undefined/yaml"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 7L12.3 6.3 8.5 10.1 8.5 1 7.5 1 7.5 10.1 3.7 6.3 3 7 8 12zM13 12v2H3v-2H2v2l0 0c0 .6.4 1 1 1h10c.6 0 1-.4 1-1l0 0v-2H13z"
              />
            </svg>
          </a>
        </div>
        <p
          class="_lastUpdate_4b0ee6"
        >
          Version created Apr 30, 2020 by ---
        </p>
        <div
          class="cds--bmrg-feature-header__footer"
        >
          <nav
            aria-label="Task template views"
            class="cds--tabs"
          >
            <ul
              class="cds--tab--list"
            >
              <a
                class="cds--tabs__nav-item cds--tabs__nav-link"
                href="/admin/task-templates/5e670a1e2d5e6a302de4f41d/1"
              >
                Overview
              </a>
              <a
                class="cds--tabs__nav-item cds--tabs__nav-link"
                href="/admin/task-templates/5e670a1e2d5e6a302de4f41d/1/yaml-editor"
              >
                Yaml
              </a>
            </ul>
          </nav>
        </div>
      </section>
      <div
        class="_buttons_4b0ee6"
      >
        <div
          class="_container_b033d1"
        >
          <div
            class="_buttonListContainer_b033d1"
          >
            <div
              class="_buttonList_b033d1"
            >
              <button
                class="_button_b033d1"
                disabled=""
              >
                <svg
                  alt="first version"
                  aria-hidden="true"
                  class="_icon_b033d1 _disabled_b033d1"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 16L24 6 25.4 7.4 16.8 16 25.4 24.6 24 26zM8 4H10V28H8z"
                  />
                </svg>
              </button>
              <button
                class="_button_b033d1"
                disabled=""
              >
                <svg
                  alt="back one version"
                  aria-hidden="true"
                  class="_icon_b033d1 _disabled_b033d1"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 8L10 3 10.7 3.7 6.4 8 10.7 12.3 10 13z"
                  />
                </svg>
              </button>
            </div>
            <p
              class="_versionText_b033d1"
            >
              Version 1
            </p>
            <div
              class="_buttonList_b033d1"
            >
              <button
                class="_button_b033d1"
                disabled=""
              >
                <svg
                  alt="forward one version"
                  aria-hidden="true"
                  class="_icon_b033d1 _disabled_b033d1"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
              </button>
              <button
                class="_button_b033d1"
                disabled=""
              >
                <svg
                  alt="last version"
                  aria-hidden="true"
                  class="_icon_b033d1 _disabled_b033d1"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 16L8 26 6.6 24.6 15.2 16 6.6 7.4 8 6zM22 4H24V28H22z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <button
          class="_resetButton_4b0ee6 cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
          disabled=""
          tabindex="0"
          type="button"
        >
           
          Reset changes
          <svg
            aria-hidden="true"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12,8c1.2,0,2,0.8,2,2c0,1.2-0.8,2-2,2H9v1h3c1.7,0,3-1.2,3-3c0-1.7-1.3-3-3-3L2.9,7l2-2L4.2,4.3L1,7.5 l3.2,3.2L4.9,10l-2-2L12,8z"
            />
          </svg>
        </button>
        <button
          class="_mainActionButton_4b0ee6 cds--btn cds--btn--md cds--btn--primary cds--btn--disabled"
          disabled=""
          tabindex="0"
          type="button"
        >
          Save...
          <svg
            aria-hidden="true"
            aria-label="Save a new version or update the current one"
            class="cds--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.9,4.6l-2.5-2.5C11.3,2.1,11.1,2,11,2H3C2.4,2,2,2.4,2,3v10c0,0.6,0.4,1,1,1h10c0.6,0,1-0.4,1-1V5 C14,4.9,13.9,4.7,13.9,4.6z M6,3h4v2H6V3z M10,13H6V9h4V13z M11,13V9c0-0.6-0.4-1-1-1H6C5.4,8,5,8.4,5,9v4H3V3h2v2c0,0.6,0.4,1,1,1 h4c0.6,0,1-0.4,1-1V3.2l2,2V13H11z"
            />
          </svg>
        </button>
      </div>
    </header>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
</body>
`;
